src/Fhir.Anonymizer.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs

```diff
@@ -38,6 +40,38 @@ public void Validate(AnonymizerConfiguration config)
                     throw new AnonymizerConfigurationErrorsException($"{method} not support.");
                 }
             }
+
+            // Check AES key size is valid (16, 24 or 32 bytes).
+            if (!string.IsNullOrEmpty(config.ParameterConfiguration?.EncryptKey))
+            {
+                using Aes aes = Aes.Create();
+                var encryptKeySize = Encoding.UTF8.GetByteCount(config.ParameterConfiguration.EncryptKey) * 8;
+                if (!IsValidKeySize(encryptKeySize, aes.LegalKeySizes))
+                {
+                    throw new AnonymizerConfigurationErrorsException($"Invalid encrypt key size : {encryptKeySize} bits! Please provide key sizes of 128, 192 or 256 bits.");
+                }
+            }
+
+        }
+
+        // The following method takes a bit length input and returns whether that length is a valid size
+        // validSizes for AES: MinSize=128, MaxSize=256, SkipSize=64
+        private bool IsValidKeySize(int bitLength, KeySizes[] validSizes)
+        {
+            int i, j;
+
+            if (validSizes == null) return false;
```

Comment:
Use parentheses to make clauses in an expression apparent.

Reply:
Thanks. Updated in the newest change.